arrayfire/array.py

```diff
@@ -20,6 +21,63 @@
 from .index import *
 from .index import _Index4
 
+_is_running_in_py_charm = "PYCHARM_HOSTED" in os.environ
+
+_display_dims_limit = None
+
+def set_display_dims_limit(*dims):
+    """
+    Sets the dimension limit after which array's data won't get
+    presented to the result of str(arr).
+
+    Default is None, which means there is no limit.
+
+    Parameters
+    ----------
+    *dims : dimension limit args
+
+    Example
+    -------
+    set_display_dims_limit(10, 10, 10, 10)
+
+    """
+    global _display_dims_limit
+    _display_dims_limit = dims
+
+def get_display_dims_limit():
+    """
+    Gets the dimension limit after which array's data won't get
+    presented to the result of str(arr).
+
+    Default is None, which means there is no limit.
+
+    Returns
+    -----------
+        - tuple of the current limit
+        - None is there is no limit
+
+    Example
+    -------
+    get_display_dims_limit()
+    # None
+    set_display_dims_limit(10, 10, 10, 10)
+    get_display_dims_limit()
+    # (10, 10, 10, 10)
+
+    """
+    return _display_dims_limit
+
+def _in_display_dims_limit(dims):
+    if _is_running_in_py_charm:
+        return False
+    print(_display_dims_limit)
+    if _display_dims_limit is not None:
+        min_dim_len = min(len(_display_dims_limit), len(dims))
+        for i in range(min_dim_len):
+            if dims[i] > _display_dims_limit[i]:
+                return False
```

Comment:
If someone sets the limits to `(20, 20)` and there's an array of size `(5, 5, 1000)`, this function returns true. May be Return false if `len(dims)` > `len(_display_dims_limit)` and then iterate only over `len(dims)`?